pkg/controllers/federatedresourcequota/federated_resource_quota_status_controller_test.go

```diff
@@ -0,0 +1,421 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federatedresourcequota
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+// TestStatusControllerReconcile tests the Reconcile function of the StatusController
+func TestStatusControllerReconcile(t *testing.T) {
+	tests := []struct {
+		name           string
+		existingObjs   []runtime.Object
+		request        reconcile.Request
+		expectedResult reconcile.Result
+		expectedError  bool
+	}{
+		{
+			name: "FederatedResourceQuota not found",
+			existingObjs: []runtime.Object{
+				&workv1alpha1.Work{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "work-1",
+						Namespace: "default",
+						Labels: map[string]string{
+							util.FederatedResourceQuotaNamespaceLabel: "default",
+							util.FederatedResourceQuotaNameLabel:      "test-quota",
+						},
+					},
+				},
+			},
+			request: reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test-quota",
+					Namespace: "default",
+				},
+			},
+			expectedResult: reconcile.Result{},
+			expectedError:  false,
+		},
+		{
+			name: "FederatedResourceQuota being deleted",
+			existingObjs: []runtime.Object{
+				&policyv1alpha1.FederatedResourceQuota{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "test-quota",
+						Namespace:         "default",
+						DeletionTimestamp: &metav1.Time{Time: time.Now()},
+						Finalizers:        []string{"karmada.io/federated-resource-quota"},
+					},
+				},
+			},
+			request: reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      "test-quota",
+					Namespace: "default",
+				},
+			},
+			expectedResult: reconcile.Result{},
+			expectedError:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			scheme := runtime.NewScheme()
+			_ = policyv1alpha1.Install(scheme)
+			_ = workv1alpha1.Install(scheme)
+			fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(tt.existingObjs...).Build()
+
+			controller := &StatusController{
+				Client:        fakeClient,
+				EventRecorder: record.NewFakeRecorder(100),
+			}
+
+			result, err := controller.Reconcile(context.Background(), tt.request)
+
+			if tt.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.expectedResult, result)
```

Comment:
The only judgment is meaningless. Can more logical judgments be added?

Reply:
Hi @XiShanYongYe-Chang , Thanks for the review. Actually the reconciliation logic and setting up manager are too complex to test and need extensive mocking.

Reply:
If so, the test function can be removed first because it is a cost to maintenance and it has not yet produced test value.

Reply:
Yeah, that will be a good step. Thanks!